source/Calamari.Common/Plumbing/Pipeline/PipelineCommand.cs

```diff
@@ -126,15 +133,31 @@ IEnumerable<Task> GetBehaviours(ILifetimeScope lifetimeScope, RunningDeployment
                 yield return ExecuteBehaviour(deployment, behaviour);
             }
 
-            yield return ExecuteBehaviour(deployment, lifetimeScope.Resolve<PackagedScriptBehaviour>(new NamedParameter("scriptFilePrefix", DeploymentStages.PostDeploy)));
-            yield return ExecuteBehaviour(deployment, lifetimeScope.Resolve<ConfiguredScriptBehaviour>(new NamedParameter("deploymentStage", DeploymentStages.PostDeploy)));
-
+            foreach (var scriptBehaviour in MaybeIncludeBehaviours<PostDeployPackagedScriptBehaviour, PostDeployConfiguredScriptBehaviour>(lifetimeScope))
+            {
+                yield return ExecuteBehaviour(deployment, scriptBehaviour);
+            }
             foreach (var behaviour in OnFinish(new OnFinishResolver(lifetimeScope)))
             {
                 yield return ExecuteBehaviour(deployment, behaviour);
             }
         }
 
+        IEnumerable<IBehaviour> MaybeIncludeBehaviours<TPackagedScriptBehaviour, TConfiguredScriptBehaviour>(ILifetimeScope lifetimeScope)
```

Comment:
Consider naming this `MaybeIncludeCustomScriptBehaviours`, since this only applies to scripts